cli: enforce build for linux/riscv64 platform

endersonmaia opened the pull request:
Since the target is always a Cartesi Machine (riscv64), cartesi build should enforce the docker build environment via `--platform` argument.

This way `Dockerfile` from cartesi/application-templates should be something like this.

**javascript**

```Dockerfile
#BUILDER
FROM --platform=$BUILDPLATFORM node:20.16.0-bookworm AS build-stage
(...)

#TARGET
FROM cartesi/node:20.16.0-jammy-slim
(...)
```

We should only need to define `--platform` when we want to run something on the host platform, for a cross-compilation or build stage that could be faster without the linux/riscv64 qemu of the docker builder.

The example above uses the `$BUILDPLATFORM`, which is recommended so it will use the host platform.

The second and last `FROM` doesn't need a `--platform` since this will be enforced by the `cartesi build`.

tuler commented on .changeset/big-ghosts-rescue.md:
```diff
@@ -0,0 +1,5 @@
+"@cartesi/cli": patch
+---
+
+enforce build for linux/riscv64 paltform
```
typo

endersonmaia replied:
even the branch name has a type :facepalm: 

I've sent a fix with push force to fix the type in the changeset and commit comment

tuler commented:
I think this will break compatibility with current Dockerfiles out there, right?

endersonmaia commented:
> I think this will break compatibility with current Dockerfiles out there, right?

Yes. :thinking: 

What do you suggest? make it a `minor` ? 

And we should also update the applications-templates.

tuler commented:
> What do you suggest? make it a `minor` ?

I think it would be a minor.

tuler commented:
As this breaks Dockerfile compatibility let's put it together with other changes that changes Dockerfile, like the v2 initiative.

endersonmaia committed:
feat(cli): enforce build for linux/riscv64 platform

since the target is always a Cartesi Machine (riscv64), cartesi build
should enforce the docker build environment via --platform argument

endersonmaia committed:
fixup! feat(cli): enforce build for linux/riscv64 platform

endersonmaia committed:
fixup! feat(cli): enforce build for linux/riscv64 platform

tuler commented on .changeset/big-ghosts-rescue.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@cartesi/cli": patch
```
in pre-release mode should we specify as major or as patch?
because it's meant to be `2.0.0-alpha.1`

endersonmaia replied:
I'm considering this particular commit as a patch. I think everything that goes into prerelease will bump the last number after `alpha` only.
